src/parser/priest/shadow/modules/core/Channeling.tsx

```diff
@@ -1,48 +1,73 @@
 import SPELLS from 'common/SPELLS';
 import CoreChanneling from 'parser/shared/modules/Channeling';
-import Events, { CastEvent, ApplyBuffEvent, RemoveBuffEvent } from 'parser/core/Events';
+import Events, { ApplyDebuffEvent, CastEvent, BeginCastEvent } from 'parser/core/Events';
+import Ability from 'parser/core/modules/Ability';
 import { SELECTED_PLAYER, Options } from 'parser/core/Analyzer';
 
 /**
  * Mind Flay and Void Torrent don't reveal in the combatlog when channeling begins and ends, this fabricates the required events so that ABC can handle it properly.
  */
+const CHANNEL_ABILITIES = [ // This used where checks are done for all channeled spriest abilities
+  SPELLS.MIND_FLAY.id,
+  SPELLS.MIND_SEAR.id,
+  SPELLS.VOID_TORRENT_TALENT.id,
+];
+
 class Channeling extends CoreChanneling {
+
+  mindBlastCastStart?: BeginCastEvent;
+
   constructor(options: Options){
     super(options);
-    this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.MIND_FLAY), this.onApplyBuff);
-    this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell([SPELLS.MIND_FLAY, SPELLS.VOID_TORRENT_TALENT]), this.onRemoveBuff);
+    this.addEventListener(Events.applydebuff.by(SELECTED_PLAYER).spell(SPELLS.MIND_FLAY), this.onApplyDebuff);
```

Comment:
something is off with the spell filters for at least onCast and onBeginCast

Reply:
Updated the filters to use all 3 spells for the filter